pi.py

```diff
@@ -1 +1,578 @@
+#             condition of an air parcel that is saturated at sea level pressure
+#
+    
+    # convert units
+    SSTK=SSTC+273.15 # SST in kelvin
+    T=T+273.15       # Temperature profile in kelvin
+    R=R*0.001        # Mixing ratio profile in gm/gm
+
+    # CHECK 1: do SSTs exceed 5C? If not, set IFL=0 and return missing PI
+    if (SSTC <= 5.0):
+        VMAX=np.nan
+        PMIN=np.nan
+        IFL=0
+        TO=np.nan
+        LNB=np.nan
+        return(VMAX,PMIN,IFL,TO,LNB)
+
+    # CHECK 2: do Temperature profiles exceed 100K? If not, set IFL=0 and return missing PI
+    if (np.min(T) <= 100):
+        VMAX=np.nan
+        PMIN=np.nan
+        IFL=0
+        TO=np.nan
+        LNB=np.nan
+        return(VMAX,PMIN,IFL,TO,LNB)
+    
+    # Set Missing mixing ratios to zero gm/gm, following Kerry's algorithm
+    R[np.isnan(R)]=0.
+    
+    # Saturated water vapor pressure
+    # from Clausius-Clapeyron relation/August-Roche-Magnus formula
+    ES0=6.112*np.exp(17.67*SSTC/(243.5+SSTC))
+
+    # Constants
+    NK=0         # level from which parcels lifted (first pressure level)
+    b=2.0        # Exponent for estimating azimuthal velocity in the eye, V=V_m(r/r_m)**b (Emanuel 1995, EQN. 25)
+    ptop=50      # Pressure below which sounding is ignored (hPa)
+    RD=287.04    # [J/kg.K] gas constant of dry air
+    EPS=RD/461.5 # [unitless] epsilon, the ratio of gas constants
+    
+    #
+    #   ***   Find environmental CAPE *** 
+    #
+    TP=T[NK]
+    RP=R[NK]
+    PP=P[NK]
+    result = cape(TP,RP,PP,T,R,P,ascent_flag,ptop,miss_handle)
+    CAPEA = result[0]
+    IFLAG = result[3]
+    # if the CAPE function tripped a flag, set the output IFL to it
+    if (IFLAG != 1):
+        IFL=int(IFLAG)
+    
+    #
+    #   ***   Begin iteration to find mimimum pressure   ***
+    #
+    
+    # set loop counter and initial condition
+    NP=0         # loop counter
+    PM=970.0
+    PMOLD=PM     # initial condition from minimum pressure
+    PNEW=0.0     # initial condition from minimum pressure
+    IFL=int(1)   # Default flag for CAPE calculation
+
+    # loop until convergence or bail out
+    while (np.abs(PNEW-PMOLD) > 0.5):
+        
+        #
+        #   ***  Find CAPE at radius of maximum winds   ***
+        #
+        TP=T[NK]
+        PP=min([PM,1000.0])
+        RP=EPS*R[NK]*MSL/(PP*(EPS+R[NK])-R[NK]*MSL)
+        result = cape(TP,RP,PP,T,R,P,ascent_flag,ptop,miss_handle)
+        CAPEM = result[0]
+        IFLAG = result[3]
+        # if the CAPE function tripped a different flag, set the output IFL to it
+        if (IFLAG != 1):
+            IFL=int(IFLAG)
+        
+        #
+        #  ***  Find saturation CAPE at radius of maximum winds    ***
+        #  *** Note that TO and LNB are found with this assumption ***
+        #
+        TP=SSTK
+        PP=min([PM,1000.0])
+        RP=0.622*ES0/(PP-ES0)
+        result = cape(TP,RP,PP,T,R,P,ascent_flag,ptop,miss_handle)
+        CAPEMS, TOMS, LNBS, IFLAG = result
```

Comment:
You could just do:
CAPEMS, TOMS, LNBS, IFLAG = cape(TP,RP,PP,T,R,P,ascent_flag,ptop,miss_handle)

Reply:
I definitely could, this is hold over that made the code a bit more readable in reference to the other cape() calls.